src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Enter.java

```diff
@@ -209,6 +209,7 @@ public Env<AttrContext> classEnv(JCClassDecl tree, Env<AttrContext> env) {
         localEnv.info.lint = null; // leave this to be filled in by Attr,
                                    // when annotations have been processed
         localEnv.info.isAnonymousDiamond = TreeInfo.isDiamond(env.tree);
+        localEnv.info.ctorPrologue = false;
```

Comment:
In my local experiments I added this to `Attr::attribClass`, but this was the other alternative, and, on a second look it probably looks like the cleanest approach.